### PR TITLE
store/tikv: update scanner lock

### PR DESCRIPTION
**Pull request by disksing**
1. Add backoff to resolve lock.
2. scanner.Next() should return a retryable error if failed to resolve a lock.

@morefreeze @qiuyesuifeng

### CODE REVIEW

File: store/tikv/lock_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	}
+	err = txn.Commit()
+	c.Assert(err, IsNil)
+}
+
+func (s *testLockSuite) TestScanLockResolve(c *C) {
+	mocktikv.BootstrapWithSingleStore(s.cluster)
+	s.putAlphabets(c)
+	s.lockKey(c, []byte("c"), []byte("c"), []byte("zz"), []byte("zz"), true)
+	s.lockKey(c, []byte("d"), []byte("d"), []byte("zz"), []byte("zz"), false)
```
Comment (morefreeze) [inline store/tikv/lock_test.go]: I think you could set different value to 'c' and 'd'. And then you could judge them in L78.

**Comment by disksing** [comment]
PTAL @morefreeze

**Comment by shenli** [comment]
LGTM

**Comment by morefreeze** [comment]
LGTM

**Pull request merged**